src/zerocoin_v3.cpp

```diff
@@ -584,8 +591,8 @@ bool ConnectBlockSigma(
             }
 
             if (!fJustCheck) {
-                pindexNew->sigmaSpentSerials.insert(serial.first);
-                sigmaState.AddSpend(serial.first);
+                pindexNew->sigmaSpentSerials.insert(serial);
+                sigmaState.AddSpend(serial.first, serial.second.denomination, serial.second.coinGroupId);
```

Comment:
AddSpend now used in 12 places, 10 of them are tests. If we implemented the change, most of the code would look like this:
`sigmaState->AddSpend(coinSerial, CSpendCoinInfo::make(denom, groupId));` or like this
`CSpendCoinInfo coinInfo = CSpendCoinInfo::make(denom, groupId);
sigmaState->AddSpend(coinSerial, coinInfo);`

As it seems to me, both would make readability worse. On the other hand, a caller of this code should not know about CSpendCoinInfo, as the CSpendCoinInfo is used for internal storage, and it may introduce unnecessary include directive which will make it slower to compile.

Let's please leave it as is to not introduce code coupling.

Reply:
Ok, but in the case of `AddMint` it is expected that the caller should know?